autoPyTorch/pipeline/components/setup/optimizer/AdamWOptimizer.py

```diff
@@ -30,13 +33,15 @@ def __init__(
         lr: float,
         beta1: float,
         beta2: float,
-        weight_decay: float,
+        use_weight_decay: bool,
+        weight_decay: float = 0,
```

Comment:
Aren't `use_weight_decay=False` and `weight_decay=0` equivalent?
We can cutout `use_weight_decay`, because we need to judge whether to use the weight decay at some points anyways and it should come at the bottom (it means in this function) to avoid the repetition.